Framework/PythonInterface/plugins/algorithms/LoadDNSLegacy.py

```diff
@@ -108,6 +123,10 @@ def PyExec(self):
             self.log().error(message)
             raise RuntimeError(message)
 
+        tmp = api.LoadEmptyInstrument(InstrumentName='DNS')
+        self.instrument = tmp.getInstrument()
+        api.DeleteWorkspace(tmp)
+
```

Comment:
It seems odd to create and delete a workspace for this but I can't think of a better way